### ISSUE TITLE
Upgrade to AssertJ 3.17.1 caused problems with assumptions

### ISSUE THREAD

**Issue by lukas-krecan** [lukas-krecan/JsonUnit#276]
We use AssertJ and JsonUnit (with JUnit4) in our project. The upgrade to AssertJ 3.17.1 caused problems with assumptions due to the [change in precedence of assumption classes](https://github.com/joel-costigliola/assertj-core/pull/1983/files#diff-ae6072599f982100da2c79beabbc6ba3R1296). Because JsonUnit depends on OpenTest4J a `org.opentest4j.TestAbortedException` is thrown by AssertJ. Junit4 does not identify that exception as an assumption error so the test fails instead of being ignored.

Maybe this investigation is related to the failed build of this PR. Hope it helps.

_Originally posted by @JoergSiebahn in https://github.com/lukas-krecan/JsonUnit/pull/274#issuecomment-686433675_

**Comment by lukas-krecan** [comment]
@JoergSiebahn Thanks for reporting it. It seems that the failures in JsonUnit test are unrelated. Can you please provide some example of the code that is failing for you?

**Comment by lukas-krecan** [comment]
[This](https://github.com/lukas-krecan/JsonUnit/pull/274/commits/20de5db5cfd01c725abe22a0358f734e6911c8c8) makes the tests pass. Will still have to investigate why the changes are needed.

**Comment by JoergSiebahn** [comment]
Thank you for the quick response.

Assumptions only fail with JUnit4. I could not find it in the classpath of JsonUnit. Maybe JUnit4 is not officially supported by JsonUnit?

However, I created a test case to reproduce the error we had in our project: https://github.com/JoergSiebahn/JsonUnit/commit/f0d62310a1864095747d49d8b321d421c8ca7154
The commit adds a new Module for tests with JUnit4. I added the same test to the test-base module where it is ignored as expected with JUnit Jupiter.

Here is the result of the failing test from the surefire reports:

```
-------------------------------------------------------------------------------
Test set: net.javacrumbs.jsonunit.test.junit4.AssumptionsWithAssertJTest
-------------------------------------------------------------------------------
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.654 s <<< FAILURE! - in net.javacrumbs.jsonunit.test.junit4.AssumptionsWithAssertJTest
net.javacrumbs.jsonunit.test.junit4.AssumptionsWithAssertJTest.shouldBeIgnoredButFails  Time elapsed: 0.6 s  <<< ERROR!
org.opentest4j.TestAbortedException: 
assumption was not met due to: 
Expecting:
 <"Test">
to be equal to:
 <"NotEqualToTest">
but was not.
	at net.javacrumbs.jsonunit.test.junit4.AssumptionsWithAssertJTest.shouldBeIgnoredButFails(AssumptionsWithAssertJTest.java:9)
Caused by: org.opentest4j.AssertionFailedError: 

Expecting:
 <"Test">
to be equal to:
 <"NotEqualToTest">
but was not.
	at net.javacrumbs.jsonunit.test.junit4.AssumptionsWithAssertJTest.shouldBeIgnoredButFails(AssumptionsWithAssertJTest.java:9)
```

**Comment by lukas-krecan** [comment]
Thanks, will take a look at it

**Comment by lukas-krecan** [comment]
Released as 2.19.0

**Comment by JoergSiebahn** [comment]
Thank you very much. Works with AssertJ 3.17.1 when excluding OpenTest4J (using Gradle, maybe not needed with Maven) and as is with AssertJ 3.17.2.
